docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+    public Dictionary<string, HighLowTemperatures> TemperatureRanges { get; set; }
+    public string [] SummaryWords { get; set; }
+}
+
+public class HighLowTemperatures
+{
+    public Temperature High { get; set; }
+    public Temperature Low { get; set; }
+}
+
+public class Temperature
+{
+    public int DegreesCelsius { get; set; }
+}
+```
+
+The JSON output is minified by default: 
+
+```json
+{"Date":"2019-08-01T00:00:00-07:00","TemperatureC":25,"Summary":"Hot","DatesAvailable":["2019-08-01T00:00:00-07:00","2019-08-02T00:00:00-07:00"],"TemperatureRanges":{"Cold":{"High":{"DegreesCelsius":20},"Low":{"DegreesCelsius":-10}},"Hot":{"High":{"DegreesCelsius":60},"Low":{"DegreesCelsius":20}}},"SummaryWords":["Cool","Windy","Humid"]}
+```
+
+The following example shows the same JSON formatted:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot",
+  "DatesAvailable": [
+    "2019-08-01T00:00:00-07:00",
+    "2019-08-02T00:00:00-07:00"
+  ],
+  "TemperatureRanges": {
+    "Cold": {
+      "High": {
+        "DegreesCelsius": 20
+      },
+      "Low": {
+        "DegreesCelsius": -10
+      }
+    },
+    "Hot": {
+      "High": {
+        "DegreesCelsius": 60
+      },
+      "Low": {
+        "DegreesCelsius": 20
+      }
+    }
+  },
+  "SummaryWords": [
+    "Cool",
+    "Windy",
+    "Humid"
+  ]
+}
+```
+
+Overloads of <xref:System.Text.Json.JsonSerializer.Serialize*> let you serialize to a `Stream`.  Async versions of the `Stream` overloads are available.
+
+### Default serialization behavior
+
+* All public properties are serialized. You can [specify properties to exclude](#exclude-properties).
+* Casing of JSON names matches the .NET names. You can [customize JSON name casing](#customize-json-names).
+* [Circular references](https://github.com/dotnet/corefx/issues/38579) are detected and exceptions thrown.
+* Fields are excluded.
+* Supported types include:
+  * .NET primitives that map to JavaScript primitives.
+  * User-defined [Plain Old CLR Objects (POCOs)](https://stackoverflow.com/questions/250001/poco-definition).
+  * One-dimensional and jagged arrays (`ArrayName[][]`).
+  * Collections such as `IList`, `ICollection`, and `IEnumerable`.
+  * `Dictionary<string,TValue>` where `TValue` is `object`, `JsonElement`, or a POCO.
+* The [default maximum depth](xref:System.Text.Json.JsonReaderOptions.MaxDepth) is 64.
+
+### Serialize to UTF-8
+
+Call [JsonSerializer.SerializeToUtf8Bytes](xref:System.Text.Json.JsonSerializer.SerializeToUtf8Bytes*):
+
+```csharp
+string json = JsonSerializer.SerializeToUtf8Bytes<WeatherForecast>(weatherForecast);
+```
+
+As an alternative, a <xref:System.Text.Json.JsonSerializer.Serialize*> overload that takes a <xref:System.Text.Json.Utf8JsonWriter> is available.
+
+Serializing to UTF-8 is about 5-10% faster than using the string-based methods. The difference is because the bytes (as UTF-8) don't need to be converted to strings (UTF-16).
+
+## How to deserialize
```

Comment:
If we do decide to add read/write to file samples, it might be good to add a note stating that no BOM is written, and on reading - the UTF-8 BOM is skipped (and a lack of any BOM is treated to mean that the data is UTF-8).